apps/common/lib/mix/tasks/namespace/release.ex

```diff
@@ -4,21 +4,42 @@ defmodule Mix.Tasks.Namespace.Release do
   """
   use Mix.Task
 
-  def run([release_version_path, remote_control_ebin]) do
+  def run(_) do
+    release = Mix.Release.from_config!(:lexical, Mix.Project.config(), [])
+    release_path = release.path
+    release_version_path = release.version_path
+
+    app_names = ~w(remote_control common common_protocol server protocol)a
+    app_paths = Enum.map(app_names, fn app ->
+      vsn = Mix.Project.in_project(app, Mix.Project.apps_paths()[app],
+        fn _ -> Mix.Project.config()[:version]
+      end)
+      Path.join([release_path, "lib", "#{app}-#{vsn}", "ebin", "#{app}.app"])
+    end)
+
     release_files = ~w(start.script lexical.rel)
-    remote_app = Path.join([remote_control_ebin, "remote_control.app"])
-    paths = [remote_app | Enum.map(release_files, &Path.join([release_version_path, &1]))]
+    script_paths = Enum.map(release_files, &Path.join([release_version_path, &1]))
+
+    paths = script_paths ++ app_paths
 
     Enum.each(paths, fn p ->
       string = File.read!(p)
       # avoid replacing directory path with negative lookbehind:
       string = String.replace(string, ~r/(?<!\/)remote_control/, "lx_remote_control")
+      string = String.replace(string, ~r/(?<!\/)common(?!_)/, "lx_common")
+      string = String.replace(string, ~r/(?<!\/)common_protocol/, "lx_common_protocol")
+      string = String.replace(string, ~r/(?<!\/)server/, "lx_server")
+      string = String.replace(string, ~r/(?<![\/_])protocol/, "lx_protocol")
```

Comment:
some of these are funny because e.g.

`protocol` should be replaced to `lx_protocol`
`common` should be replaced to `lx_common`
but
`common_protocol` should be replaced to `lx_common_protocol`

it should *not* replace to `lx_common_lx_protocol`

Reply:
also this is a bit of a "hit and miss" job, I am assuming I don't need to add the `proto` app, because this worked fine. But through usage we find ze bugz.

Reply:
why no pipeline?